app/services/insights.js

```diff
@@ -0,0 +1,342 @@
+
+      // Run additional custom serializer if relevant
+      if (typeof currentOptions.customSerialize === 'function') {
+        try { [newKey, newVal] = currentOptions.customSerialize(newKey, newVal); } catch (e) {}
+      }
+
+      return [newKey, newVal];
+    });
+    currentMetric.plotData = plotData;
+
+    // Split into labels and values - works better for some charting libraries
+    currentMetric.plotLabels = currentMetric.plotData.map(([key, val]) => key);
+    currentMetric.plotValues = currentMetric.plotData.map(([key, val]) => val);
+
+    // Calculate total
+    currentMetric.total = currentMetric.plotValues.reduce((acc, val) => acc + val, 0);
+    total += currentMetric.total;
+
+    // Calculate average if requested. Uses total calculated above.
+    if (currentOptions.calcAvg) {
+      const filtered = currentMetric.plotValues.filter(v => v !== 0);
+      currentMetric.average = filtered.length === 0 ? 0 : currentMetric.total / filtered.length;
+      avgs.push(currentMetric.average);
+    }
+  });
+
+  // Calculate grand total / average of all metrics
+  serData.total = total;
+  if (currentOptions.calcAvg) {
+    let avgsTotal = avgs.reduce((a, v) => a + v, 0);
+    serData.average = avgs.length === 0 ? 0 : avgsTotal / avgs.length;
+  }
+
+  return serData;
+}
+
+// These aggregator functions are for aggregating data when there is a key collision, i.e. when
+// there are multiple values for a given time. This happens on the owner page because repos are
+// the main thing that the insights app aggregates data around
+function getAggregator(aggName) {
+  return {
+    sum: sumAggregator,
+    max: maxAggregator,
+    avg: avgAggregator,
+    count: countAggregator,
+  }[aggName];
+}
+
+function sumAggregator(map, name, time, value) {
+  if (map[name].hasOwnProperty(time)) {
+    map[name][time] += value;
+  } else {
+    map[name][time] = value;
+  }
+  return map;
+}
+
+function maxAggregator(map, name, time, value) {
+  if (map[name].hasOwnProperty(time)) {
+    if (value > map[name][time]) {
+      map[name][time] = value;
+    }
+  } else {
+    map[name][time] = value;
+  }
+  return map;
+}
+
+function avgAggregator(map, name, time, value) {
+  // if (value === 0) { return map; }
+  if (map[name].hasOwnProperty(time) && typeof map[name][time] !== 'number') {
+    map[name][time][0]++;
+    map[name][time][1] += value;
+  } else {
+    map[name][time] = [1, value];
+  }
+  return map;
+}
+
+function countAggregator(map, name, time, value) {
+  if (map[name].hasOwnProperty(time)) {
+    map[name][time]++;
+  } else {
+    map[name][time] = 1;
+  }
+  return map;
+}
+
+// These serializer functions are for putting point data into a format to be read by the
+// charting components
+function getSerializer(serializerName) {
+  const serializers = {
+    avg: avgSerializer,
+    default: defaultSerializer,
+  };
+  return serializers[serializerName] || serializers['default'];
```

Comment:
Just to avoid using `default` (as it's a keyword in JS) this could be something like:
```javascript
const serializers = {
  avg: avgSerializer
};
return serializers[serializerName] || defaultSerializer;
```
Though, it's entirely optional 😃

Reply:
Ahh right totally! That makes sense to me, change implemented :)